Automattic/wp-calypso
# Checkout: Move checkout types to wpcom-checkout package

#### Changes proposed in this Pull Request

This moves many TypeScript types from calypso checkout to the `@automattic/wpcom-checkout` npm package. The npm package has type safety, whereas calypso does not, so anything that uses types is safer if moved to the package. Eventually I'd like to move a lot of the checkout functions to this package but by moving the types first it will make those changes considerably smaller, and moving the types alone (even if there are bugs) should have no impact on calypso since the latter doesn't do any type checking.

While this doesn't move all the types from checkout, it moves a good chunk of them and should be a good starting place. The more we move at once the harder this will be to review and the more time it will take. Since any such moves are likely to generate conflicts, I think it's best to get started.

#### Types moved by this PR

```
WPCOMTransactionEndpointCart
WPCOMTransactionEndpointResponse
Purchase
TransactionRequest
WPCOMTransactionEndpoint
WPCOMTransactionEndpointRequestPayload
WPCOMTransactionEndpointPaymentDetails
PossiblyCompleteDomainContactDetails
DomainContactDetailsErrors
CaDomainContactExtraDetailsErrors
UkDomainContactExtraDetailsErrors
FrDomainContactExtraDetailsErrors
PayPalExpressEndpoint
PayPalExpressEndpointRequestPayload
PayPalExpressEndpointResponse
WPCOMCart
CheckoutPaymentMethodSlug
WPCOMPaymentMethod
ContactDetailsType
ManagedContactDetailsShape
ManagedContactDetailsTldExtraFieldsShape
ManagedContactDetails
ManagedContactDetailsErrors
ManagedContactDetailsUpdate
ManagedContactDetailsRequiredMask
ManagedValue
WpcomStoreState
FailedPurchase
ManagedContactDetailsUpdaters
SignupValidationResponse
DomainContactValidationRequest
GSuiteContactValidationRequest
DomainContactValidationRequestExtraFields
DomainContactValidationResponse
```

#### Testing instructions

This should have no functional effect since it moves only type definitions. Any chance that it breaks checkout should be covered by automated tests, but just to be sure, open checkout and verify that it loads correctly.
